sky/skylet/providers/lambda_cloud/node_provider.py

```diff
@@ -164,31 +170,39 @@ def internal_ip(self, node_id: str) -> str:
     def create_node(self, node_config: Dict[str, Any], tags: Dict[str, str],
                     count: int) -> None:
         """Creates a number of nodes within the namespace."""
-        assert count == 1, count  # Only support 1-node clusters for now
-
-        # get the tags
+        # Get tags
         config_tags = node_config.get('tags', {}).copy()
         config_tags.update(tags)
         config_tags[TAG_RAY_CLUSTER_NAME] = self.cluster_name
 
-        # create the node
+        # Create nodes
         ttype = node_config['InstanceType']
         region = self.provider_config['region']
-        vm_list = self.lambda_client.create_instances(instance_type=ttype,
-                                                      region=region,
-                                                      quantity=1,
-                                                      name=self.cluster_name)
-        assert len(vm_list) == 1, len(vm_list)
-        vm_id = vm_list[0]
-        self.metadata[vm_id] = {'tags': config_tags}
-
-        # Wait for booting to finish
-        # TODO(ewzeng): For multi-node, launch all vms first and then wait.
+        if config_tags[TAG_RAY_NODE_KIND] == NODE_KIND_HEAD:
+            name = f'{self.cluster_name}-head'
+        else:
+            name = f'{self.cluster_name}-worker'
+        # Lambda launch api only supports launching one node at a time,
+        # so we do a loop. Remove loop when launch api allows quantity > 1
+        booting_list = []
+        for _ in range(count):
+            vm_id = self.lambda_client.create_instances(instance_type=ttype,
+                                                        region=region,
+                                                        quantity=1,
+                                                        name=name)[0]
```

Comment:
Can we have some rate limit error handling / retrying for the place we call the lambda API? One possible problem that I could image here is we run `sky launch -c test1 --cloud lambda` and `sky launch -c test2 --cloud lambda` in parallel, and they could fail each other.

I am thinking if we can have the retry logic built-in the `LambdaClient`, i.e. if the API rate limit is reached, retry the function call with some backoff.

Reply:
Hmm, let me think about this. In this case, the backoff time would require some randomness (or else the `test1` and `test2` would keep failing each other).

The first launch should always succeed (we use a different api to test availability that doesn't have a harsh rate limit)...